### PR TITLE
Rspec

### PR DESCRIPTION
**Pull request by julianpalladino**
**Summary**
- Used rspec to test models and controllers

**Trello card**
https://trello.com/c/7SVhhKir/12-crear-tests-con-rspec-de-modelos-y-controllers

### CODE REVIEW

File: spec/controllers/article_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+        end
+
+        it "renders the index template" do
+          get :index
+          expect(response).to render_template("index")
+        end
+      end
+
+
+      DatabaseCleaner.start
```
Comment (mdesanti) [inline spec/controllers/article_controller_spec.rb]: Why is this here?

File: spec/controllers/article_controller_spec.rb [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+      let!(:articles) { FactoryGirl.create_list(:article, 180) }
+      it "loads all of the articles into @articles" do
+        get :index
+
+        expect(assigns(:articles)).to match_array(articles)
+      end
+
+      DatabaseCleaner.clean
```
Comment (mdesanti) [inline spec/controllers/article_controller_spec.rb]: This shouldn't be here. Check [this](https://github.com/Wolox/rails-bootstrap/blob/master/spec/support/spec_helper.rb#L25) to see how it is implemented.

You can configure Rspec to execute some code before each test and after each test. In this way you can avoid adding this at the beginning of each context

File: spec/controllers/article_controller_spec.rb [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+        get :new
+
+        expect(response).to render_template("new")
+      end
+    end
+
+    context "when user is not logged in" do
+      it "renders sign in template" do
+        get :new
+        expect(response).to redirect_to("http://test.host/auth/sign_in")
```
Comment (mdesanti) [inline spec/controllers/article_controller_spec.rb]: Having the URL wired here makes the test less flexible. Let's suppose that tomorrow you want to change the path for the user to login. This test will break although you haven't changed the behaviour of the `new` method. You could do something like this

`expect(response).to redirect_to(new_session_path)`

File: spec/controllers/article_controller_spec.rb (hunk 1 above)
Comment (julianpalladino) [inline spec/controllers/article_controller_spec.rb]: What do you mean specifically?

Comment (mdesanti) [inline spec/controllers/article_controller_spec.rb]: Check the comment below

**Comment by mdesanti** [comment]
@julianpalladino rebase this branch first and then i'll check it . It has many changes that I think that are not from this branch

**Comment by mdesanti** [comment]
:+1:
